compiler/rustc_session/src/config/cfg.rs

```diff
@@ -142,6 +142,10 @@ pub(crate) fn disallow_cfgs(sess: &Session, user_cfgs: &Cfg) {
             | (sym::target_has_atomic, Some(_))
             | (sym::target_has_atomic_equal_alignment, Some(_))
             | (sym::target_has_atomic_load_store, Some(_))
+            | (sym::target_has_reliable_f16, Some(_))
+            | (sym::target_has_reliable_f16_math, Some(_))
+            | (sym::target_has_reliable_f128, Some(_))
+            | (sym::target_has_reliable_f128_math, Some(_))
```

Comment:
Unless I'm missing something they don't expect a value, so we should also disallow value-less cfgs.
```suggestion
            | (sym::target_has_reliable_f16, None | Some(_))
            | (sym::target_has_reliable_f16_math, None | Some(_))
            | (sym::target_has_reliable_f128, None | Some(_))
            | (sym::target_has_reliable_f128_math, None | Some(_))
```